lib/README.md

```diff
@@ -0,0 +1,29 @@
+# lowRISC Nix Libraries
+
+## The Documentation Library
+
+The `doc` library contains `buildMdbookSite` and `hasExts`.
+The former is a helper function for creating mdBook site builds.
+An example of it's use is below.
+
+```nix
+let
+  fs = pkgs.lib.fileset;
+in {
+  docs_build = doc.buildMdbookSite {
+    pname = "documentation";
+    version = "0.0.1";
+    nativeBuildInputs = [pkgs.python311];
+    src = fs.toSource {
+      root = ./.;
+      fileset = fs.unions [
+        ./book.toml
+        (fs.fileFilter (doc.hasExts ["md" "svg" "png"]) ./.)
```

Comment:
Would it make sense to create still have a default filter for book.toml and these extensions?

E.g.
```nix
fileset = fs.unions [
  (doc.mdbookFilter ./.)
  ./util/mdbook
]
```
?

Reply:
It would need to be:

```nix
fileset = fs.unions [
  (doc.mdbookFilter ./.)
  ./util/mdbook
  ./util/mdbook_wavejson.py
]
```

I tried to do a `doc.mdbookFilter` but with no luck.